rust/connlib/libs/common/src/session.rs

```
@@ -394,7 +390,12 @@ where
         // So always yield and if you spawn a blocking tasks rewrite this.
         // Furthermore, we will depend on Drop impls to do the list above so,
         // implement them :)
-        *runtime.lock() = None;
+        // if there's no receiver the runtime is already stopped
+        // there's an edge case where this is called before the thread is listening for stop threads.
+        // but I believe in that case the channel will be in a signaled state achieving the same result
+        if let Err(err) = runtime_stopper.try_send(()) {
```

Could this be a oneshot instead? Theoretically, an `mpsc` could not have any space which is an error case that would be good to avoid here. With a oneshot, the only possible error case is "disconnected" which would allow you to either ignore the error or print a better message.

---

the problem is that in a `oneshot` `send` consumes `self` so I'd need to wrap it in an `Arc<Mutex<Option<OneShot<T>>>>` or similar which makes it a whole more cumbersome to use and this function should tear down the whole runtime anyways so it should never run out of buffer.